src/derivatives.jl

```diff
@@ -0,0 +1,55 @@
+@doc raw"""
+    Y = geodesic_derivative(M, p, X, t; γt = geodesic(M, p, X, t))
+    geodesic_derivative!(M, Y, p, X, t; γt = geodesic(M, p, X, t))
+
+Evaluate the derivative of the geodesic ``γ(t)`` with ``γ_{p,X}(0) = p`` and ``\dot γ_{p,X}(0) = X`` at ``t``.
+The formula reads
+
+```math
+\dot γ(t) = \mathcal P_{γ(t) \gets p} X
+```
+
+where $\mathcal P$ denotes the parallel transport.
+This computation can also be done in-place of ``Y``.
+
+# Optional Parameters
+
+* `γt` – (`geodesic(M, p, X, t)`) the point on the geodesic at ``t``.
+  This way if the point was computed earlier it can be resued here.
+"""
+function geodesic_derivative(M, p, X, t; q = geodesic(M, p, X, t))
```

Comment:
Shouldn't it be `derivative_geodesic` in the new naming scheme?

Reply:
I'd also suggest adding `t::Number` here.

Reply:
> Shouldn't it be `derivative_geodesic` in the new naming scheme?

would be a possibility, my decision here was since we write `f‘` in math to go for `f_derviative` (while for `Df` we write `derivative_f`. Does that sound reasonable?

Reply:
Hm, OK, I think that's fine, though I'm going to confuse it anyway. Could you just explain the naming scheme somewhere in docs?

Reply:
I added a paragraph to the docs.

The docs in general could be a bit longer, I might work on that in a next PR. For example splitting the library of actual derivatives from the available backends.

Reply:
Thanks, I think this description is perfectly fine for this PR.